### PR TITLE
Stripe -  Synch User/Invoices Reorg

### PR DESCRIPTION
**Pull request by ClaudeROSSI**
Checking LIVE MODE flag to switch off some features while we are still in beta mode.

- The purpose is to make it easier to test the payment flow on SLF without any impct on regular users 
- The user synchronization logic as been simplified to avoid some ping-pong between eMobility ad STRIPE.
- Still need to review  the invoice synchronization logic - current implementation is slow and most of the time not necessary anymore
- Includes as well some additional checks to make sure customerID is properly set and  consistent
- Deletion of users on the STRIPE side is not supported anymore! - will be done later

### CODE REVIEW

File: src/integration/billing/stripe/StripeBillingIntegration.ts [hunk 1]
```diff
@@ -1194,102 +1155,86 @@ export default class StripeBillingIntegration extends BillingIntegration<StripeB
...
-        // ---------------------------------------------------------------------------------------
-        // This should not happen - The customerID
-        // The customerID refers to something which does not exists anymore in the STRIPE account
-        // May happen when billing settings are changed to point to a different STRIPE account
-        // ---------------------------------------------------------------------------------------
-        if (strictMode) {
+        // Gets the STRIPE Customer
+        const customer: Stripe.Customer | Stripe.DeletedCustomer = await this.stripe.customers.retrieve(customerID);
+        // Check for the deletion flag
+        const deleted = (customer.deleted) ? true : false; // ACHTUNG - STRIPE type definition is wrong!
```
Comment (cabch) [inline src/integration/billing/stripe/StripeBillingIntegration.ts:1209]: for me it works. tried assigning myself a deleted cus_ and returned a Stripe.DeletedCustomer

File: src/integration/billing/stripe/StripeBillingIntegration.ts [hunk 2]
```diff
@@ -589,39 +516,37 @@ export default class StripeBillingIntegration extends BillingIntegration<StripeB
     // Check Stripe
     await this.checkConnection();
     // Check billing data consistency
-    // necessary to check stripe user exists here ?
-    // if we don't verify here stripe answers we don't have payment methods - no error is thrown !!
-    if (!user?.billingData?.customerID || await this._isDeletedInStripe(user?.billingData?.customerID)) {
```
Comment (cabch) [inline src/integration/billing/stripe/StripeBillingIntegration.ts:594]: don't you want to also delete the method ?

Comment (cabch) [reply src/integration/billing/stripe/StripeBillingIntegration.ts:594]: will be deleted

Comment (ClaudeROSSI) [reply src/integration/billing/stripe/StripeBillingIntegration.ts:594]: done

**Comment by ClaudeROSSI** [comment]
@LucasBrazi06 : This one includes a lot of changes but mainly some refactoring.

- synchronizeUsers and synchronizedInvoices are switched off behind the LIVE MODE flag which is set to OFF to let us test the billing beta version on SLF.

**Review by cabch — APPROVED** [review]

**Pull request merged**